orchestrator-salt/src/main/resources/salt-common/salt/fluent/init.sls

```diff
@@ -86,6 +90,7 @@ install_fluentd_plugins:
       - group
       - mode
 
+{% if not fluent.fluent_prewarm_role in grains.get('roles', []) %}
```

Comment:
my problem here is that i described in https://github.com/hortonworks/cloudbreak/pull/8457#discussion_r452182829
It will mean many extra support hours for us as we will hide the root cause of version incompatibility based on the pre-warm image that is used.

my suggestion for this instead pre warm role check, create a new field in image catalog (metadata), same as package_versions, (let's say fluentd_plugin_versions), then you can pass those image values as salt pillars, and you can check the versions in fluent/settings.sls -> if any of the versions are different (between CB salt script constant <-> pillar), then configure and run this script.

cc @horadla23 @keyki

Reply:
I'm not comfortable with the solution of extending the metadata of image in catalog (since it requires lot of modification in various repositories thus it is error prune), but if that is the best we can do, I'm not against it. 

To do that, you should:
- add logic in cloudbreak-images repository to generate map of fluent_plugin_versions map into metadata which is passed to cb-image-catalog-cli
- extend model in cb-image-catalog-cli and update logic regarding metadata/image copy
- extend models in cloudbreak
- implement logics regarding that map in cloudbreak

please if it possible do not modify package-versions, since it is responsible for change image/OS upgrade, i suggest a new field for this

Reply:
Had explicitly skipped the option of trying to add these to the image catalog. There's multiple modules associated with fluent which would need to be versioned, and the image catalog changes are far more disruptive.
The current solution is quite a bit simpler and seems adequate. Quoting from the other PR

> 1. if we have new image in production but a new CB is not deployed - version conflict (plugins)
The image, if fluent changes are required, would have changed the version on the fluent role. e.g. "fluent_prewarmed_v2" instead of "fluent_prewarmed_v1". Since CB has not been upgraded, the checks would fail (i.e. result is false), and regular attempts would be made to install fluent (i.e. the >= check will execute and effectively be a no-op). No different from what happens today.

> 2. New CB deployed, a node uses an older image (it happens often) - version conflict (plugins)
In this case, CB would have changed the fluent role in settings.sls from e.g. "fluent_prewarmed_v1" to "fluent_prewarmed_v2". Older images would still have "fluent_prewarmed_v1". The check would fail (i.e. result is false), and all the package checks would execute as they do today.

If both CB and the image are upgraded - we get the benefit of a faster install, which will likely be the majority scenario.

Reply:
@oleewere what is the agreement here?

Reply:
@lacikaaa i guess we keep this as is in the PR, although the check should be more smart, (so if the version is higher on the image, that should not be a problem, so we need to lookup in the grains and get the version part of fluent_prewarmed_v1 value, and compare with that)